app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -507,9 +509,17 @@ private void processUpdateMessages(List<EventsBranch> updateEvents) {
             @Override
             public void run() {
                 RecyclerMessageAdapter adapter = mActivity.getCurrentMessageList().getAdapter();
+                int i = 0;
                 for (int id : messageIds) {
                     // notify adapter data item changed
-                    adapter.notifyItemChanged(adapter.getItemIndex(id));
+                    int index = adapter.getItemIndex(id);
+                    adapter.notifyItemChanged(index);
+                    RecyclerView.ViewHolder viewHolderForAdapterPosition = mActivity.getCurrentMessageList().getRecyclerView().findViewHolderForAdapterPosition(index);
+                    if (viewHolderForAdapterPosition instanceof MessageHolder) {
+                        UpdateMessageWrapper updateEvent = (UpdateMessageWrapper) updateEvents.get(i++);
```

Comment:
I think an assumption is made here that all updateEvents will have a corresponding messageId in the arraylist but I believe that isn't true https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java#L499 for all scenarios.